### DISCUSSION TITLE
Keep Conversation only for a week.

### DISCUSSION THREAD

**Discussion by jigarjain709** [chatwoot/chatwoot#5931] [General]
Hello,

Is there a way to delete all conversations older than a week? 

We are using chatwoot mainly for Whatsapp Business API. We have no use of chats/conversations older than 1 week without deleting the contact. So, is there a way that I can delete these older conversations along with the media (through UI or Postgress CLI) without affecting the integrity of the system?

**Comment by glauberssoutodossantos** [comment]
you can create a routine that deletes messages within the conversation, but keeps the conversation active.

That way you don't have to delete everything and you can make it more dynamic.

**Reply by jigarjain709** [reply]
Thanks, @glauberssoutodossantos.

Is there an option in the super admin panel to create the routine or will I have to schedule a cron job to execute a script?

**Reply by glauberssoutodossantos** [reply]
Hi,

You need to create a routine outside chatwoot :(
